include/RAJA/policy/cuda/policy.hpp

```diff
@@ -66,7 +68,7 @@ struct Dim3z {
 /// Segment execution policies
 ///
 
-struct cuda_exec_base {
+struct cuda_exec_base : public forall_policy {
```

Comment:
This is interesting. Essentially here we say that reduce (and taskgraph?) policies will never inherit from backend_exec_base policies, that is, everything which inherits from cuda_exec_base must be a forall_policy. I don't know that it's wrong, I just think it should be discussed

edit: the more I look at this, the more I think you might have the basis of a good refactoring, note how some backends have an exec_base, some (like seq) just have an exec forall policy.